.gitlab-ci.yml

```diff
@@ -1,14 +1,139 @@
+  <<: *tox_3_test
+
+tox_34_check:
+  stage: test
+  image: fedora:29
+  variables:
+    TOXENV: 'unit_py3_4'
+    PYTHON: 'python3.4'
+  before_script:
+    - dnf install -y --refresh python34 tox libxml2-devel libxslt-devel gcc enchant make gcc which genisoimage xz
+  <<: *tox_3_test
+
+tox_27_check:
+  stage: test
+  image: opensuse/tumbleweed
+  before_script:
+    - zypper --non-interactive refresh
+    - zypper --non-interactive install libxml2-devel libxslt-devel gcc python2 python2-devel python2-virtualenv python2-pip enchant which mkisofs xz gzip
+  script:
+    - virtualenv .env2
+    - source .env2/bin/activate
+    - pip install -r .virtualenv.dev-requirements.txt
+    - python2 setup.py develop
+    - tox -e unit_py2_7 "-n $(nproc)"
+
+OpenSUSE:
+  stage: test
+  image: opensuse/tumbleweed
+  before_script:
+    - zypper --non-interactive refresh
+    - zypper --non-interactive install python3 python3-devel python3-pip
+    # requirements
+    - zypper --non-interactive install python3-docopt python3-lxml python3-PyYAML python3-xattr python3-six python3-future python3-requests
+    # dev-requirements
+    - zypper --non-interactive install python3-pytest python3-pytest-cov python3-mock python3-flake8 python3-Sphinx python3-sphinx_rtd_theme python3-pyenchant python3-tox make gcc which mkisofs xz gzip
+    # pytest-xdist from the repos is broken, use the one from PyPI instead
+    - pip3 install sphinxcontrib-spelling pytest-xdist
+  <<: *distro_test
+
+Fedora:
+  stage: test
+  image: fedora:29
+  <<: *fedora_install_deps
+  <<: *distro_test
+
+build_doc:
+  stage: doc
+  image: fedora:29
+  <<: *fedora_install_deps
+  script:
+    - dnf -y install latexmk texlive-cmap texlive-metafont texlive-ec texlive-babel-english texlive-fncychap texlive-fancyhdr texlive-titlesec texlive-tabulary texlive-framed texlive-wrapfig texlive-parskip texlive-upquote texlive-capt-of texlive-needspace texlive-makeindex texlive-times texlive-helvetic texlive-courier texlive-gsftopk texlive-updmap-map texlive-dvips
+    - cd doc
+    - make html SPHINXBUILD=/usr/bin/sphinx-build-3
+    - make linkcheck SPHINXBUILD=/usr/bin/sphinx-build-3
+    - make man SPHINXBUILD=/usr/bin/sphinx-build-3
+    - make latexpdf SPHINXBUILD=/usr/bin/sphinx-build-3
+    # FIXME: there is a lot of warnings here, but they are not fatal
+    # => remove this maybe
+    - make spell SPHINXBUILD=/usr/bin/sphinx-build-3
+  artifacts:
+    paths:
+      - doc/build/
+
+RPM:
+  stage: package
+  image: fedora:29
+  before_script:
+    - dnf -y install fedpkg mock perl-TimeDate which
+  script:
+    - 'sed -i "s|build: clean tox|build:|" Makefile'
+    - make build
+    - cd dist
+    - fedpkg --release f29 srpm
+    - export SRC_RPM=$(ls *src.rpm)
+    - mock --old-chroot -r /etc/mock/opensuse-tumbleweed-x86_64.cfg $SRC_RPM
+    - mock --old-chroot -r /etc/mock/fedora-29-x86_64.cfg $SRC_RPM
```

Comment:
Mock can build using a spec file directly if you give it `--sourcedir` and `--spec` instead of the `$SRC_RPM` argument. I advise that the `--sourcedir` only contain the sources that the spec would reference to make it faster for copying into the build environment.

Reply:
This option is no longer/not yet present? Mock can build SRPMs via very similar options though.

Reply:
I have used the SRPM build option, so now fedpkg is no longer required.